dynamicio/handlers/s3/file.py

```diff
@@ -0,0 +1,99 @@
+"""S3 File Handlers for dynamicio."""
+
+from copy import deepcopy
+from pathlib import Path
+from threading import Lock
+from typing import Any, Callable, Dict
+
+import boto3  # type: ignore
+import pandas as pd  # type: ignore
+from pydantic import Field
+
+from dynamicio.base import BaseResource
+from dynamicio.handlers.s3.contexts import s3_named_file_reader
+from dynamicio.inject import check_injections, inject
+
+hdf_lock = Lock()
+
+
+class BaseS3Resource(BaseResource):
+    """S3 Resource - kwargs dict is forwarded.
+
+    S3 Resources try to read data directly into memory if no_disk_space is set to True.
+    Otherwise, will use a temporary file.
+
+    Attributes:
+        bucket: The name of the bucket.
+        path: The path to the file.
+        kwargs: A dictionary of kwargs to be passed to the read/write function.
+        no_disk_space: If True, will read data directly into memory. (uses s3fs + fsspec)
+    """
+
+    bucket: str
+    path: Path
+    kwargs: Dict[str, Any] = {}
+
+    no_disk_space: bool = False
```

Comment:
I think `force_in_memory_write: bool = False` could be slightly more suggestive than `no_disk_space`. Not a strong opinion

Reply:
I copied this from v4 and had the same hunch. I will change this.